core-kotlin-modules/core-kotlin-10/src/test/kotlin/com/baeldung/variableshadowing/VariableShadowingUnitTest.kt

```diff
@@ -0,0 +1,188 @@
+@file:Suppress("UNUSED_VARIABLE", "UNUSED_ANONYMOUS_PARAMETER")
+
+package com.baeldung.variableshadowing
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+class VariableShadowingUnitTest{
+
+    @Test
+    fun `Top-level variable shadowing`(){
+        val number = 10 // Top-level variable
+
+        fun upNumber() : Int { // top-level function
+            val number = 20 // shadowing top-level variable
+            return number
+        }
+
+        assertEquals(20, upNumber())
+        assertEquals(10, number)
+    }
+
+    @Test
+    fun `shadowing class member`(){
+        class Car {
+            val speed: Int = 100
+
+            fun upSpeed() : Int {
+                val speed = speed * 2 // Shadowing class member named 'speed'
+                return speed
+            }
+        }
+
+        assertEquals(100, Car().speed)
+        assertEquals(200, Car().upSpeed())
+    }
+
+    @Test
+    fun `local variable shadowing`(){
+        fun calculateTotalPrice(discount: Int) {
+            val discount = discount + 10 // Shadowing the parameter 'discount'
+            assertEquals(30, discount)
+
+            val price = 100 // local variable
+            val discountRate = 0.1
+
+            fun applyDiscount(price: Int): Double { // Nested function with parameter named 'price'
+                val discountRate = 0.2  // shadowing the outer variable discountRate
+                return price * (1 - discountRate) // 'price' here refers to the parameter, not the outer variable
+            }
+
+            assertEquals(80.0, applyDiscount(price))
+        }
+
+        calculateTotalPrice(20)
+    }
+
+
+    @Test
+    fun `shadowing in loop`(){
+        val numbers = listOf(1, 2, 3, 4, 5)
+
+        // shadowing in loop
+        for (number in numbers) {
+            val number = number * 2 // Shadowing the loop variable 'number'
+        }
+    }
+
+    @Test
+    fun `shadowing in extension`(){
+        val numbers = listOf(1, 2, 3, 4, 5)
+
+        assertEquals(15, numbers.sum())
+
+        fun List<Int>.sum(): Int { // shadowing built-in function sum()
+            var sum = 0
+            this.forEach { sum += it * 2 }
+            return sum
+        }
+
+        assertEquals(30, numbers.sum())
+    }
+
+    @Test
+    fun `shadowing in lambda`(){
+        val numbers = listOf(1, 2, 3, 4, 5)
+
+        var sum = 0
+
+        numbers.forEach { number ->
+            val number = 0 // shadowing value
+            sum += number
+        }
+
+        assertEquals(0, sum)
+
+        val lambda = { number : Int ->
+            val number = 1// Local variable in lambda
+        }
+    }
+
+
+    @Test
+    fun `solution to avoid shadowing`(){
```

Comment:
This should also be split, so each of the "solution to X" tests corresponds to one of the "shadowing" tests above

And let's order the tests like:
`shadowing class member`
`solution to avoid shadowing class member`
`shadowing in lambda`
`solution to avoid shadowing in lamda`
...